python/pyspark/sql/types.py

```diff
@@ -187,8 +187,12 @@ def needConversion(self):
 
     def toInternal(self, dt):
         if dt is not None:
-            seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
-                       else time.mktime(dt.timetuple()))
+            # Avoiding the invalid range of years (100-1899) for mktime in Python < 3
+            if dt.year > 1899 or dt.year <= 100:
+                seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
+                           else time.mktime(dt.timetuple()))
+            else:
+                seconds = calendar.timegm(dt.utctimetuple())
```

Comment:
Well, this was the previous functionality, I'm not 100% sure of the rationale that was used to do it this way either. If we don't have `tzinfo` and use `dt.timetuple` instead of `utctimetule` I guess it's using locale time via the underlying C implementation for `mktime`, or that's what the Python docs seem to suggest, but it's not that clear. So I'm not convinced we need to offset, since it should be a platform/locale dependent offset